requests/utils.py

```diff
@@ -169,12 +169,15 @@ def super_len(o):
 def get_netrc_auth(url, raise_errors=False):
     """Returns the Requests tuple auth for a given url from netrc."""
 
+    netrc_file = os.environ.get('NETRC')
+    netrc_locations = (netrc_file,) if netrc_file is not None else NETRC_FILES
+
     try:
         from netrc import netrc, NetrcParseError
 
         netrc_path = None
 
-        for f in NETRC_FILES:
+        for f in netrc_locations:
             try:
                 loc = os.path.expanduser('~/{}'.format(f))
```

Comment:
This doesn't look quite right. Are all `NETRC` paths relative to the users home directory? I think we only want this formatting for the `NETRC_FILES`.

Reply:
Actually, it looks like the last PR has [similar feedback](https://github.com/psf/requests/pull/4339#discussion_r144723124). If we can get that or something similar incorporated, I think we're set.

Reply:
very true... how's this then?

Reply:
I can also do it exactly as suggested in previous PR... let me know